deepinv/tests/test_utils.py

```diff
@@ -900,6 +900,30 @@ def test_prepare_images(x, y, x_net, x_nl, rescale_mode):
         assert all(
             isinstance(title, str) for title in titles
         ), "All titles should be strings."
+        assert len(imgs) == len(
+            titles
+        ), "Number of images should match number of titles."
+        assert len(imgs) == sum(v is not None for v in [x, y, x_net, x_nl])
```

Comment:
It fails because the test to include `y` is not just `y is not None`, but: `y is not None and x is not None and y.shape == x.shape`.

An ugly but working fix (feel free to improve it!):

```suggestion
      assert len(imgs) == sum(v is not None for v in [x, x_net, x_nl]) + int(
            y is not None and x is not None and y.shape == x.shape
      )
```

Reply:
Also maybe add an assert message if you can think of a good one! Happy to leave it that way if not